### PR TITLE
feat(ci): add ci for auto release

### PR DESCRIPTION
**Pull request by asjdf**
First of all sorry for the late reply.

In this pr, I hope to introduce an automated release process for the project, aiming to reduce the workload of the release, and at the same time make it easier for other developers to freely choose and switch between the development version, alpha version, beta version, and official version.

Below I will introduce the specific usage details of the automatic release introduced by this PR.

1. If someone wants to make a new feature, he should make a new branch which name can't be `dev`, `alpha` or `beta`.
2. Commit with the correct title which should follow the rule down below.
3. After development, he should make a pull request to merge the branch into the `dev` branch (or merge directly).
4. After merging, the release workflow will auto-release a new dev version of this project.
5. We can do the same progress to release the `main`, `alpha` or `beta` version.

| Commit message                                                                                                                                                                                   | Release type                                                                                                    |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------- |
| `fix(pencil): stop graphite breaking when too much pressure applied`                                                                                                                             | ~~Patch~~ Fix Release                                                                                           |
| `feat(pencil): add 'graphiteWidth' option`                                                                                                                                                       | ~~Minor~~ Feature Release                                                                                       |
| `perf(pencil): remove graphiteWidth option`<br><br>`BREAKING CHANGE: The graphiteWidth option has been removed.`<br>`The default graphite width of 10mm is always used for performance reasons.` | ~~Major~~ Breaking Release <br /> (Note that the `BREAKING CHANGE: ` token must be in the footer of the commit) |

Thanks for your review ❤.

Best regards,
Asjdf

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#      - uses: codecov/codecov-action@v3.1.1
+#        with:
+#          file: ./profile.cov
+#          name: codecov-go
+  release:
+    runs-on: ubuntu-22.04
+    needs: [test]
+#    if: ${{ github.event_name == 'workflow_dispatch' || github.ref == 'refs/heads/main' }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (s8sg) [inline .github/workflows/release.yaml:44]: Do we need to define it or its auto populated ?

Comment (asjdf) [reply .github/workflows/release.yaml:44]: This environment variable is automatically set by Github.

If some problems occur, you may need to adjust workflow's token permissions in the repo Settings.

Comment (giorgiozoppi) [reply .github/workflows/release.yaml:44]: Also can we use gitversion for semantic versioning? https://github.com/GitTools/GitVersion

**Review by s8sg — APPROVED** [review]
looks good

**Pull request merged**